src/linearmodel.jl

```diff
@@ -187,41 +212,275 @@ function discrete_jacobian!(::Type{DiscreteSystemQuadrature}, ∇f, model::Discr
+    if is_affine   
+        _ltv_affine(name, n, m, N, :DiscreteLTV)
+    else
+        _ltv_non_affine(name, n, m, N, :DiscreteLTV)
+    end
+end
 
-    continuous_system = zero(SizedMatrix{n+m, n+m})
-    continuous_system[1:n, 1:n] .= A_c
-    continuous_system[1:n, n .+ (1:m)] .= B_c
+"""
+    @create_continuous_ltv(name, n, m, N, is_affine=false)
+
+This macro can be used to conveniently create a ContinuousLTV model subtype and model instance.
+In order to use the resulting model, use the `set_A!`, `set_B!`, `set_times!`, and `set_d!`
+functions to add in your system matrices. This macro must be called from a top level scope.
+
+Creates a struct that looks like:
+```julia
+struct (name){TA, TB, Td, T} <: ContinuousLTV
+    A::Vector{TA}
+    B::Vector{TB}
+    d::Vector{Td}
+    times::Vector{T}
+end
+```
+All functions necessary to modify/use the struct are predefined by the macro. The macro defines a 
+default constructor using `SMatrix` with all zeros. This can be called simply with `(name)()`.
+If `is_affine = false` then the d term is not defined in the struct.
+"""
+macro create_continuous_ltv(name, n, m, N, is_affine=false)
+    if is_affine   
+        _ltv_affine(name, n, m, N, :ContinuousLTV)
+    else
+        _ltv_non_affine(name, n, m, N, :ContinuousLTV)
+    end
+end
 
-    discrete_system = exp(continuous_system*dt)
-    A_d = discrete_system[StaticArrays.SUnitRange(1,n), StaticArrays.SUnitRange(1,n)]
-    B_d = discrete_system[StaticArrays.SUnitRange(1,n), StaticArrays.SUnitRange(n+1,n+m)]
+function _ltv_non_affine(name, n, m, N, supertype)
+    struct_exp = quote
+        struct ($name){TA, TB, T} <: ($supertype)
+            A::Vector{TA}
+            B::Vector{TB}
+            times::Vector{T}
+        end
+    end
+    function_def = quote
+        RobotDynamics.is_affine(::($name)) = Val(false)
+        RobotDynamics.control_dim(::($name)) = $m
+        RobotDynamics.state_dim(::($name)) = $n
+        RobotDynamics.get_A(model::($name), k::Integer) = model.A[k]
+        RobotDynamics.get_B(model::($name), k::Integer) = model.B[k]
+        RobotDynamics.get_times(model::($name)) = model.times
+        RobotDynamics.set_A!(model::($name), A::AbstractArray, k::Integer) = model.A[k] = A
+        RobotDynamics.set_B!(model::($name), B::AbstractArray, k::Integer) = model.B[k] = B
+        RobotDynamics.set_times!(model::($name), times::AbstractVector) = model.times .= times
+
+        function ($name)()
+            A_vec = [@SMatrix zeros($n, $n) for _ = 1:($N-1)]
+            B_vec = [@SMatrix zeros($n, $m) for _ = 1:($N-1)]
+            times = zeros($N)
+
+            TA = eltype(A_vec)
+            TB = eltype(B_vec)
+            T = eltype(times)
+
+            return ($name){TA, TB, T}(A_vec, B_vec, times)
+        end
+    end
+
+    esc(Expr(:toplevel,
+            struct_exp,
+            function_def))
+end
 
-    return (A_d, B_d)
+function _ltv_affine(name, n, m, N, supertype)
```

Comment:
There's a lot of redundant code in here. For example, I think it would be pretty easy to combine the affine and non-affine functions in one. Just have a little bit of branching for the few things that are different rather than copying a bunch of code. It makes it easier to maintain and reduces bloat.